### PR TITLE
feat(ui): admin page

### PR DESCRIPTION
**Pull request by Ethan-Chew**

### CODE REVIEW

File: components/admin/event/event-checkedin-users.vue [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<script setup lang="ts">
+    import { f7List, f7ListItem, f7Card, f7CardContent, f7CardHeader } from 'framework7-vue';
+    const props = defineProps(["event"])
```
Comment (qin-guan) [inline components/admin/event/event-checkedin-users.vue]: Can use typed props?

File: components/admin/event/event-checkedin-users.vue [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+</script>
+
+<template>
+    <f7Card>
+        <f7CardHeader>
+            <span class="font-semibold text-2xl">Attendees</span>
+        </f7CardHeader>
+        <f7CardContent>
+            <f7List v-for="attendee in props.event.attendees" media-list dividers-ios strong inset>
```
Comment (qin-guan) [inline components/admin/event/event-checkedin-users.vue]: v-for need key attr

https://vuejs.org/api/built-in-special-attributes.html#key

File: components/admin/event/event-information.vue [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<script setup lang="ts">
+import { f7Card, f7CardHeader, f7Chip, f7CardContent } from 'framework7-vue';
+
+const props = defineProps(["event"])
```
Comment (qin-guan) [inline components/admin/event/event-information.vue]: same as above can use typed prop

File: components/admin/event/event-information.vue [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                        <span class="font-bold text-3xl md:text-4xl">{{ props.event.name }}</span>
+                        <div>
+                            <div class="flex-inline shrink grow-0 p-1 px-4 border border-solid b-rounded-3">
+                                {{ `0 / ${event.attendees.length} Attendees Checked-In` }}
+                            </div>
+                        </div>
+                    </div>
+                    <div class="text-lg flex flex-col">
+                        <span><b>{{ convertEpochToSGT(props.event.startDateTime) }}</b> to <b>{{ convertEpochToSGT(props.event.endDateTime) }}</b></span>
+                        <span><b>Location:</b> {{ props.event.location }}</span>
```
Comment (qin-guan) [inline components/admin/event/event-information.vue]: prefer font-semibold or font-bold

File: components/admin/event/event-information.vue [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    <f7Card>
+        <f7CardContent class="p-0! lg-rounded!">
+            <div class="relative">
+                <img :src="props.event.badgeImage" class="object-cover w-full max-h-60 object-top lg-rounded!">
+                <div class="p-10 absolute z-10 inset-0 flex flex-col bg-gray-800/[0.3] space-y-6 backdrop-blur-sm lg-rounded!">
+                    <div class="space-y-3 flex flex-col">
+                        <span class="font-bold text-3xl md:text-4xl">{{ props.event.name }}</span>
+                        <div>
+                            <div class="flex-inline shrink grow-0 p-1 px-4 border border-solid b-rounded-3">
+                                {{ `0 / ${event.attendees.length} Attendees Checked-In` }}
```
Comment (qin-guan) [inline components/admin/event/event-information.vue]: nit: 

0 / {{ event.attendees.length }} looks cleaner

File: components/admin/event/upload-attendees.vue [hunk 6]
```diff
@@ -0,0 +1,126 @@
+<script setup lang="ts">
+    import { f7List, f7ListItem, f7Card, f7CardContent, f7CardHeader, f7Button } from 'framework7-vue';
+    import { parse, ParseResult } from 'papaparse'
+    import { ref } from 'vue'
```
Comment (qin-guan) [inline components/admin/event/upload-attendees.vue]: unnecessary import

File: components/admin/home/create-event-popup.vue [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+})
+
+const state = reactive({
+    pending: false,
+    isCreated: false,
+    isError: false,
+    newEventId: "",
+})
+
+async function createEvent() {
```
Comment (qin-guan) [inline components/admin/home/create-event-popup.vue:28]: use vue-query mutation

File: components/admin/home/delete-event-popup.vue [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+const props = defineProps(["event"])
+
+const state = reactive({
+    pending: false,
+    verification: "",
+    msg: "",
+    isDeleted: false,
+})
+
+async function deleteEvent() {
```
Comment (qin-guan) [inline components/admin/home/delete-event-popup.vue]: same, use mutation

File: components/admin/home/events-table.vue [hunk 9]
```diff
@@ -11,7 +31,7 @@ const { data: events } = useEvents()
       <f7Card v-for="event in events" :key="event.id">
         <f7CardHeader>
           <div class="flex flex-col items-start space-y-4">
-            <f7Chip outline :text="`0 / ${event.attendees.length}`" />
+            <f7Chip outline :text="`0 / ${event.attendees !== undefined ? event.attendees.length : ''}`" />
```
Comment (qin-guan) [inline components/admin/home/events-table.vue]: Can use optional chaining

File: components/admin/home/events-table.vue [hunk 10]
```diff
@@ -21,15 +41,27 @@ const { data: events } = useEvents()
           {{ event.description }}
         </f7CardContent>
         <f7CardFooter>
-          <f7Button tonal>
+          <f7Button tonal @click="() => selectedEvent = event">
```
Comment (qin-guan) [inline components/admin/home/events-table.vue]: Should not need cb

File: components/admin/home/events-table.vue [hunk 11]
```diff
@@ -21,15 +41,27 @@ const { data: events } = useEvents()
...
+          </f7Button>
         </f7CardFooter>
       </f7Card>
     </div>
 
     <div class="hidden md:inline">
       <div class="data-table data-table-init card">
+        <div class="card-header">
+          <div class="data-table-title">Events</div>
+          <div></div>
```
Comment (qin-guan) [inline components/admin/home/events-table.vue]: Is this required?

File: components/admin/home/events-table.vue [hunk 12]
```diff
@@ -21,15 +41,27 @@ const { data: events } = useEvents()
...
+          <div class="data-table-actions">
+            <a class="button popup-open icon-only" data-popup=".create-event-popup">
+              <i class="icon f7-icons"> plus_circle </i>
```
Comment (qin-guan) [inline components/admin/home/events-table.vue]: Can use f7Icon component

File: components/admin/home/update-event-popup.vue [hunk 13]
```diff
@@ -0,0 +1,69 @@
+<script setup lang="ts">
+import { f7Popup, f7Page, f7Navbar, f7NavRight, f7Link, f7BlockTitle, f7Block, f7List, f7Button, f7ListInput } from 'framework7-vue';
+import { DeprecationTypes } from 'nuxt/dist/app/compat/capi';
+
+const props = defineProps(["event"])
+
+function epochToISO(epochTime: number) {
```
Comment (qin-guan) [inline components/admin/home/update-event-popup.vue]: Please use dayjs

File: components/admin/home/update-event-popup.vue [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+    endDateTime: epochToISO(props.event.endDateTime),
+})
+
+const state = reactive({
+    pending: false,
+    msg: ""
+})
+
+async function updateEvent() {
+    console.log(updatedValues)
```
Comment (qin-guan) [inline components/admin/home/update-event-popup.vue]: Can remove?

File: components/admin/home/update-event-popup.vue [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+                    <f7Link popup-close>Cancel</f7Link>
+                </f7NavRight>
+            </f7Navbar>
+            <f7Page>
+                <f7BlockTitle large>{{ `Update Event: ${ props.event.name }` }}</f7BlockTitle>
+                <f7Block>
+                    Some content goes here
+                </f7Block>
+                <f7List form @submit.prevent="updateEvent">
+                    {{ console.log(epochToISO(props.event.endDateTime)) }}
```
Comment (qin-guan) [inline components/admin/home/update-event-popup.vue]: Can remove?

File: components/admin/home/update-event-popup.vue [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+                    <f7ListInput label="Event ID" v-model:value="props.event.id" disabled />
+                    <f7ListInput label="Event Name" v-model:value="updatedValues.name" :placeholder="props.event.name" />
+                    <f7ListInput label="Event Description" v-model:value="updatedValues.description" :placeholder="props.event.description" />
+                    <f7ListInput label="Event Location" v-model:value="updatedValues.location" :placeholder="props.event.location" />
+                    <f7ListInput label="Image URL" v-model:value="updatedValues.badgeImage" :placeholder="props.event.url" type="url" />
+                    <f7ListInput label="Start Date and Time" v-model:value="updatedValues.startDateTime" :placeholder="epochToISO(props.event.startDateTime)" type="datetime-local" />
+                    <f7ListInput label="End Date and Time" v-model:value="updatedValues.endDateTime" :placeholder="epochToISO(props.event.endDateTime)" type="datetime-local" />
+
+                    <f7List inset>
```
Comment (qin-guan) [inline components/admin/home/update-event-popup.vue]: Can remove?

File: pages/admin/[...slug].vue [hunk 17]
```diff
@@ -26,6 +26,11 @@ const appRoutes = [
     path: '/admin/members',
     component: AdminMembersPage,
   },
+  {
+    name: "event-details",
+    path: '/admin/event/:id',
+    component: AdminEventSlug
```
Comment (qin-guan) [inline pages/admin/[...slug].vue]: Maybe better name would be AdminEventDetails

File: shared/types.ts [hunk 18]
```diff
@@ -2,6 +2,7 @@ import type { InferSelectModel } from 'drizzle-orm'
 import type { events, news, users } from '~/server/db/schema'
 
 export type User = InferSelectModel<typeof users>
+export type ImportedUser = Omit<User, 'firebaseId' | 'id'>
```
Comment (qin-guan) [inline shared/types.ts]: Is this typed used anywhere other than on front end

**Comment by qin-guan** [comment]
Oh no typecheck failed :(

File: shared/types.ts (hunk 18 above)
Comment (Ethan-Chew) [reply shared/types.ts]: Nope, should I add it somewhere else?

Comment (qin-guan) [reply shared/types.ts]: Yeah i think can just inline

File: components/admin/home/events-table.vue (hunk 11 above)
Comment (Ethan-Chew) [reply components/admin/home/events-table.vue]: Wanted to make it a more defined section

File: components/admin/event/detail.vue [hunk 19]
```diff
@@ -0,0 +1,37 @@
...
+import type { Router } from 'framework7/types'
+import { typeIsEventWithAttendees } from '~/composables/event'
+
+const props = defineProps<{
+  f7route: Router.Route
+}>()
+const { data: user } = useUser()
+const { data: event } = useEvent(props.f7route.params.id!)
+
+const showForbidden = computed(() => {
```
Comment (qin-guan) [inline components/admin/event/detail.vue]: I think this logic is actually shifted out to the main slug page component already, so should be safe to remove

File: components/admin/event/event-information.vue [hunk 20]
```diff
@@ -0,0 +1,53 @@
+<script setup lang="ts">
+import { f7Card, f7CardContent } from 'framework7-vue'
+
+const props = defineProps(['event'])
```
Comment (qin-guan) [inline components/admin/event/event-information.vue]: Typed props (:

**Comment by qin-guan** [comment]
Closed in favor of #7

**Pull request closed without merging**